php_phongo.c

```diff
@@ -1111,8 +1165,17 @@ static bool php_phongo_apply_rp_options_to_uri(mongoc_uri_t *uri, bson_t *option
 		mongoc_read_prefs_set_mode(new_rp, MONGOC_READ_SECONDARY_PREFERRED);
 	}
 
-	if (bson_iter_init_find_case(&iter, options, MONGOC_URI_READPREFERENCE) && BSON_ITER_HOLDS_UTF8(&iter)) {
-		const char *str = bson_iter_utf8(&iter, NULL);
+	if (bson_iter_init_find_case(&iter, options, MONGOC_URI_READPREFERENCE)) {
```

Comment:
It occurred to me while working on this PR that the driver handles duplicate options with different character casing very poorly. For instance, if the options array contained both `readpreference` and `readPreference`, this logic would only pick up the first matching key and ignore subsequent duplicates. Ideally, we would raise some exception for duplicates or simply require a single form to be used (e.g. "readPreference"). The latter would be a BC break and would serve little benefit if duplicate detection is implemented.

I believe libmongoc's handling of URI strings processes each parameter as it appears in the string so that the last value overrides any previous occurrences.

I've opened [PHPC-991](https://jira.mongodb.org/browse/PHPC-991) to track this.